src/SdfEntityCreator.cc

```diff
@@ -379,6 +380,32 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Light *_light)
   this->dataPtr->ecm->CreateComponent(lightEntity,
       components::Name(_light->Name()));
 
+  if (_light->Type() == sdf::LightType::POINT)
+  {
+    this->dataPtr->ecm->CreateComponent(lightEntity,
+        components::LightType(std::string("Point")));
+  }
+  else if (_light->Type() == sdf::LightType::DIRECTIONAL)
+  {
+    this->dataPtr->ecm->CreateComponent(lightEntity,
+        components::LightType(std::string("Directional")));
+  }
+  else if (_light->Type() == sdf::LightType::SPOT)
+  {
+    this->dataPtr->ecm->CreateComponent(lightEntity,
+        components::LightType(std::string("Spot")));
+  }
+
+  // Entity
+  Entity visualEntity = this->dataPtr->ecm->CreateEntity();
```

Comment:
I'm not sure that creating a visual entity for the light is the best approach here. The light visuals exist for visualization purposes. They're the 3D representation of a light, like the orange boxes are the 3D representation of collisions. They're not simulated entities with their own properties. I recommend removing these entities.

Reply:
The idea behind this it to let the user know that there is a visual attached to the light. The same way we include visual or collisions to a model. But I fine removing it

Reply:
I think the word "visual" has been a source of confusion since Gazebo classic. There are visuals in the SDF, which are those inside a `<visual>` tag, and there are visuals for the rendering scene, which can represent various different SDF tags, like models, links, collisions, lights... This often leads to very confusing terminology like [visual visual](https://github.com/osrf/gazebo/blob/92248a75919603293fbe67bc2716eb59010e7eec/test/integration/visual_pose.cc#L63-L64) for the "rendering visual of an SDF visual".

This is to say that the "visual entities" on Ignition Gazebo correspond to the SDF visuals. While what you're adding here is a visualization of a `<light>` entity. Since they're meant for visualization purposes, they're not part of the simulation, and the ECM doesn't need to know about them. That's the same way as we have link visuals, model visuals and collision visuals - all which are not represented as separate entities.

Let me know if this makes things clearer.

Reply:
Thank you for the clarification, then we don't need this portion of code. It was removed